### PR TITLE
feat: Expose Setting for GRPC Channel-Level Compression at Client Side

### PR DESCRIPTION
**Pull request by geetu040**
### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you installed `pre-commit` with `pip3 install pre-commit` and set up hooks with `pre-commit install`?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### Issue

This PR refers to this issue #87 . It enables client to set GZip, grpc compression algorithm. By default no algorithm is applied.

### Following code was modified

- 1st commit: functions in qdrant_client/connection.py
	- get_channel
	- get_async_channel
- 2nd commit: remote classes
	- class QdrantRemote in qdrant_client/qdrant_remote.py
	- class AsyncQdrantRemote in qdrant_client/async_qdrant_remote.py
- 3rd commit: client classes and grpc compression enum
	- class QdrantClient in qdrant_client/qdrant_client.py
	- class AsyncQdrantClient in qdrant_client/async_qdrant_client.py
	- qdrant_client/grpc/\_\_init\_\_.py

### Usage

Gzip compression algorithm
```python
from qdrant_client import QdrantClient
from qdrant_client.grpc import Compression

client = QdrantClient(
	prefer_grpc=True,
	grpc_compression=Compression.Gzip
)
```

No compression algorithm
```python
from qdrant_client import QdrantClient
from qdrant_client.grpc import Compression

# client = QdrantClient(prefer_grpc=True)
# client = QdrantClient(prefer_grpc=True, grpc_compression=None)
client = QdrantClient(prefer_grpc=True, grpc_compression=Compression.NoCompression)
```

### CODE REVIEW

**Comment by joein** [comment]
Hi @geetu040 

Thanks for the contribution!

Do you think it would be possible to add some tests?

**Comment by geetu040** [comment]
I have added the test cases

**Comment by geetu040** [comment]
This PR Closes #87

**Comment by geetu040** [comment]
Hi @joein 
If you think the changes are all good to go, can you merge them?

**Comment by joein** [comment]
Hey

Thanks for the contribution and sorry for the late response

Actually we would not like to extend client's signature with this parameter, since the signature is already large.
We haven't seen a big demand for this option either.
Thus, we would like to keep it in **kwargs

**Comment by geetu040** [comment]
Hey @joein 
I have moved grpc_compression parameter from client's signature to **kwargs

File: qdrant_client/grpc/__init__.py [hunk 1]
```diff
@@ -5,3 +5,26 @@
...
 from .snapshots_service_pb2_grpc import *
+
+
+# Compression Enum Implementation
+
+import enum
+import grpc
+
+@enum.unique
+class Compression(enum.IntEnum):
```
Comment (joein) [inline qdrant_client/grpc/__init__.py:16]: I don't think we need a separate structure just to exclude `Deflate`, can we just use `isinstance` in init or deeper?

**Review by joein — CHANGES_REQUESTED** [review]

File: qdrant_client/grpc/__init__.py (hunk 1 above)
Comment (geetu040) [reply qdrant_client/grpc/__init__.py:16]: @joein 
`grpc` module in `qdrant_client` does not support `Compression` enum
```py
from qdrant_client import grpc

grpc.Compression
# AttributeError: module 'qdrant_client.grpc' has no attribute 'Compression'. Did you mean: 'CompressionRatio'?
```

Thus we would either need to explicitly import Compression in `grpc/__init__.py` or keep the `grpc_compression` input as string. What would you suggest?

File: qdrant_client/async_qdrant_remote.py [hunk 2]
```diff
@@ -112,10 +113,14 @@ def __init__(
             self._rest_headers["api-key"] = api_key
             self._grpc_headers.append(("api-key", api_key))
         grpc_compression = kwargs.pop("grpc_compression", None)
```
Comment (joein) [inline qdrant_client/async_qdrant_remote.py:115]: could you add Optional[grpc.Compression] please?

Comment (geetu040) [reply qdrant_client/async_qdrant_remote.py:115]: should it be like this?
```python
grpc_compression: Optional[grpc.Compression] = kwargs.pop("grpc_compression", None)
```

Comment (joein) [reply qdrant_client/async_qdrant_remote.py:115]: yeap

**Comment by joein** [comment]
eh, found a discrepancy between pre-commit setup and async-client-generator, should be fixed by now

**Comment by joein** [comment]
@geetu040 thanks for the contribution!

**Pull request merged**